presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/IcebergMetadataManager.java

```diff
@@ -0,0 +1,73 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.util;
+
+import com.facebook.presto.hive.HdfsEnvironment;
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.facebook.presto.iceberg.IcebergConfig;
+import com.facebook.presto.iceberg.IcebergMetadata;
+import com.facebook.presto.iceberg.IcebergResourceFactory;
+import com.facebook.presto.iceberg.IcebergTransactionManager;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import org.apache.iceberg.Table;
+
+import javax.inject.Inject;
+
+import static com.facebook.presto.iceberg.IcebergUtil.getIcebergTable;
+import static com.facebook.presto.iceberg.util.IcebergPrestoModelConverters.toIcebergTableIdentifier;
+
+public class IcebergMetadataManager
+{
+    private final IcebergResourceFactory resourceFactory;
+    private final IcebergConfig config;
+    private final IcebergTransactionManager transactionManager;
+    private final HdfsEnvironment hdfsEnvironment;
+
+    @Inject
+    IcebergMetadataManager(
+            IcebergConfig config,
+            IcebergResourceFactory resourceFactory,
+            IcebergTransactionManager transactionManager,
+            HdfsEnvironment hdfsEnvironment)
+    {
+        this.config = config;
+        this.resourceFactory = resourceFactory;
+        this.transactionManager = transactionManager;
+        this.hdfsEnvironment = hdfsEnvironment;
+    }
+
+    public Table getTable(ConnectorSession session, ConnectorTransactionHandle transaction, SchemaTableName schemaTableName)
+    {
+        if (config.isNativeMode()) {
+            return resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(schemaTableName));
+        }
+        else {
+            ExtendedHiveMetastore metastore = ((IcebergMetadata) transactionManager.get(transaction)).getMetastore();
+            return getIcebergTable(metastore, hdfsEnvironment, session, schemaTableName);
+        }
+    }
+
+    public Table getColumnHandles(ConnectorSession session, ConnectorTransactionHandle transaction, SchemaTableName schemaTableName)
+    {
+        if (config.isNativeMode()) {
+            return resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(schemaTableName));
+        }
+        else {
+            ExtendedHiveMetastore metastore = ((IcebergMetadata) transactionManager.get(transaction)).getMetastore();
+            return getIcebergTable(metastore, hdfsEnvironment, session, schemaTableName);
+        }
+    }
```

Comment:
Looks like the two functions here are exactly the same excepts for the names. Are there any specific reasons that you want them to be two functions?

Reply:
my bad, this new file is for another refactoring PR, actually it's not used by this PR, I will remove this file from this PR.  Thanks